### PR TITLE
osemgrep ci: report findings

### PR DESCRIPTION
**Pull request by hannesm**
this is at the moment reporting null to the complete endpoint, but does report to findings (some partial mocked data -- see TODO items).

test plan: make core

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: src/osemgrep/networking/Scan_helper.ml [hunk 1]
```diff
@@ -115,3 +115,106 @@ let report_failure ~dry_run ~token ~scan_id exit_code =
     | Error (code, msg) ->
         Error
           ("API server returned " ^ string_of_int code ^ ", this error: " ^ msg)
+
+(* TODO the server reply would be specified in ATD *)
```
Comment (aryx) [inline src/osemgrep/networking/Scan_helper.ml]: do you know to which endpoints this refers too? Might be nice to add also in the comment the endpoint.

File: src/osemgrep/networking/Scan_helper.ml [hunk 2]
```diff
@@ -115,3 +115,106 @@ let report_failure ~dry_run ~token ~scan_id exit_code =
...
+                  (JSON.string_of_json (JSON.Object xs))))
+    | json ->
+        Logs.err (fun m -> m "Not a json object %s" (JSON.string_of_json json))
+  with
+  | e ->
+      Logs.err (fun m ->
+          m "Failed to decode server reply as json %s: %s"
+            (Printexc.to_string e) data)
+
+(* TODO the server reply should be specified in ATD *)
```
Comment (aryx) [inline src/osemgrep/networking/Scan_helper.ml]: same here

**Comment by aryx** [comment]
pre-commit failure (semgrep jsonnet) in     src/osemgrep/cli_ci/Ci_subcommand.ml

**Comment by aryx** [comment]
Not sure why we don't get the actual error message ...
We're now dogfooding osemgrep in precommit, so this might be a bug in osemgrep :(
In any case you can run 'make check' locally to get the same error.

**Review by aryx — CHANGES_REQUESTED** [review]
Looks good, just need to fix precommit

**Comment by hannesm** [comment]
addressed the review comments, and pre-check

File: src/osemgrep/networking/Scan_helper.ml (hunk 2 above)
Comment (hannesm) [reply src/osemgrep/networking/Scan_helper.ml]: done in [d8aaaf4](https://github.com/returntocorp/semgrep/pull/8168/commits/d8aaaf41a50b5dc72a71fb776e75a8422f4b87d5)

File: src/osemgrep/networking/Scan_helper.ml (hunk 1 above)
Comment (hannesm) [reply src/osemgrep/networking/Scan_helper.ml]: done in [d8aaaf4](https://github.com/returntocorp/semgrep/pull/8168/commits/d8aaaf41a50b5dc72a71fb776e75a8422f4b87d5)

**Comment by hannesm** [comment]
> Not sure why we don't get the actual error message ... We're now dogfooding osemgrep in precommit, so this might be a bug in osemgrep :( In any case you can run 'make check' locally to get the same error.

Strange. The command line is morally `osemgrep --config semgrep.jsonnet --error --skip-unknown-extensions .` -- which locally reports the findings as pysemgrep (on stdout)...

But what is even stranger is that there's no report of "N findings", only ever 0 findings are reported. :/

And additionally, the exit code is 2 for osemgrep, not 1 which would be expected.

EDIT: there's as well duplicate headings and tables... and on a Linux machine running `pre-commit run semgrep -a` results in the same behaviour: only the filename is printed, but not the finding... but I can't reproduce by invoking osemgrep on the shell, only if executed via pre-commit...

**Comment by aryx** [comment]
pre-commit is a bit weird because it does run many semgrep (or osemgrep) commands with different batch of files, which is why we have lots of similar boilerplate output repeated (we should really have a --precommit flag to optimize output for precommit).

**Comment by aryx** [comment]
I think there's a way to run precommit in verbose mode so you can see the different commands that are executed.

**Pull request merged**